src/commonMain/kotlin/fr/acinq/lightning/serialization/v2/Serialization.kt

```diff
@@ -0,0 +1,244 @@
+            subclass(UpdateAddHtlc.serializer())
+            subclass(UpdateFailHtlc.serializer())
+            subclass(UpdateFailMalformedHtlc.serializer())
+            subclass(UpdateFee.serializer())
+            subclass(UpdateFulfillHtlc.serializer())
+        }
+    }
+
+    private val tlvSerializersModule = SerializersModule {
+        polymorphic(Tlv::class) {
+            subclass(ChannelTlv.UpfrontShutdownScript.serializer())
+            subclass(ChannelTlv.ChannelVersionTlv.serializer())
+            subclass(InitTlv.Networks.serializer())
+            subclass(OnionTlv.AmountToForward.serializer())
+            subclass(OnionTlv.OutgoingCltv.serializer())
+            subclass(OnionTlv.OutgoingChannelId.serializer())
+            subclass(OnionTlv.PaymentData.serializer())
+            subclass(OnionTlv.InvoiceFeatures.serializer())
+            subclass(OnionTlv.OutgoingNodeId.serializer())
+            subclass(OnionTlv.InvoiceRoutingInfo.serializer())
+            subclass(OnionTlv.TrampolineOnion.serializer())
+            subclass(GenericTlv.serializer())
+        }
+    }
+
+    private val serializersModule = SerializersModule {
+        polymorphic(ChannelStateWithCommitments::class) {
+            subclass(Normal::class)
+            subclass(WaitForFundingConfirmed::class)
+            subclass(WaitForFundingLocked::class)
+            subclass(WaitForRemotePublishFutureCommitment::class)
+            subclass(ShuttingDown::class)
+            subclass(Negotiating::class)
+            subclass(Closing::class)
+            subclass(ErrorInformationLeak::class)
+        }
+    }
+
+    private val serializationModules = SerializersModule {
+        include(tlvSerializersModule)
+        include(updateSerializersModule)
+        include(SerializersModule {
+            contextual(ByteVector64KSerializer)
+            contextual(ByteVector32KSerializer)
+            contextual(ByteVectorKSerializer)
+            contextual(SatoshiKSerializer)
+            contextual(PrivateKeyKSerializer)
+            contextual(PublicKeyKSerializer)
+            contextual(OutPointKSerializer)
+            contextual(TxInKSerializer)
+            contextual(TxOutKSerializer)
+            contextual(TransactionKSerializer)
+            contextual(BlockHeaderKSerializer)
+        })
+    }
+
+    // used by the "test node" JSON API
+    val lightningSerializersModule = SerializersModule {
+        include(serializersModule)
+        include(tlvSerializersModule)
+        include(updateSerializersModule)
+    }
+
+    @OptIn(ExperimentalSerializationApi::class)
+    fun serialize(state: ChannelStateWithCommitments): ByteArray {
+        val output = ByteArrayOutput()
+        val encoder = DataOutputEncoder(output)
+        encoder.encodeSerializableValue(ChannelStateWithCommitments.serializer(), state)
+        val bytes = output.toByteArray()
+        val versioned = SerializedData(version = 1, data = bytes.toByteVector())
```

Comment:
Because our initial plan was to keep using the same serializers for base types (byte arrays, transactions, ....) and just change channel state serializers, which did not work out. But I guess it may makes senses anyway (debugging, ...)